utils/upload.go

```diff
@@ -72,6 +75,38 @@ func UploadToCloudStorage(uploader Uploader, path string) {
 	uploader.Upload(walker)
 }
 
+func GetPreSignedURL(videoId string) string {
+	cloudSession := configs.GetCloudSession()
+
+	containerName := constants.CLOUD_CONTAINER_NAME_TEMP
+
+	if configs.EnvVar[configs.GCP_PROJECT_ID] != "" {
+		return getPresignedGCPURL(&GcpUploader{
+			ContainerName: containerName,
+			VideoId:       videoId,
+			Client:        cloudSession.GCPSession,
+		})
+	}
+
+	if configs.EnvVar[configs.AWS_ACCESS_KEY_ID] != "" {
+		return getPresignedAWSURL(AwsUploader{
+			ContainerName: containerName,
+			VideoId:       videoId,
+			Session:       cloudSession.AWSSession,
+		})
+	}
+
+	if configs.EnvVar[configs.AZURE_ACCESS_KEY] != "" {
+		return getPresignedAzureURL(AzureUploader{
+			ContainerName: containerName,
+			VideoId:       videoId,
```

Comment:
Use switch-case whenever multiple if-else are involved

```go
switch configs.EnvVar[configs.GCP_PROJECT_ID] {
case "":
    return getPresignedGCPURL(&GcpUploader{
        ContainerName: containerName,
        VideoId:       videoId,
        Client:        cloudSession.GCPSession,
    })
case "":
    return getPresignedAWSURL(AwsUploader{
        ContainerName: containerName,
        VideoId:       videoId,
        Session:       cloudSession.AWSSession,
    })
case "":
    return getPresignedAzureURL(AzureUploader{
        ContainerName: containerName,
        VideoId:       videoId,
        Credential:    cloudSession.AzureSession,
    })
default:
    return ""
}
```

Reply:
here, we are checking != "", and we can't negate values in switch case, thats why used if